File: net/structure-in-rte/structure_in_rte_implement_link_resolver.cs
using Kontent.Ai.Delivery.ContentItems.RichText.Resolution;

// Define URL patterns for resolving content item links by content type
// Available placeholders: {codename}, {type}, {urlslug}, {id}
var linkResolver = DefaultResolvers.UrlPatternResolver(new Dictionary<string, string>
{
    ["article"] = "/articles/{urlslug}"
});

// For other means of resolving links, see SDK docs:
// https://github.com/kontent-ai/delivery-sdk-net/blob/master/docs/rich-text-customization.md#content-item-link-resolvers

File: ruby/structure-in-rte/structure_in_rte_implement_link_resolver.rb
# Tip: Find more about Ruby SDKs at https://kontent.ai/learn/ruby
require 'delivery-sdk-ruby'

link_resolver = Kontent::Ai::Delivery::Resolvers::ContentLinkResolver.new(lambda do |link|
        # Link is available
        return "/articles/#{link.url_slug}" if link.type.eql? 'article'
      end, lambda do |id|
        # Link is broken
        return "/notfound?id=#{id}"
      end)
